docs/new-architecture-backward-turbomodule.md

```diff
@@ -0,0 +1,699 @@
+
+<figure>
+  <img src="docs/assets/Turbomodules/OldArchApp.png" />
+  <figcaption>The simulator running the app</figcaption>
+</figure>
+
+But, if we tap on the button, nothing happens. That's because the `Core Location` system requires some keys in the plist. Let's do the following:
+
+1. Open the `ios/OldArchitectureApp.xcworkspace`.
+1. Select the project in the project navigator.
+1. Select the info tab.
+1. Scroll down and add the following two keys:
+   - `NSLocationWhenInUseUsageDescription`
+   - `NSLocationAlwaysAndWhenInUseUsageDescription`
+1. Add any description you want for each of these.
+
+Go back to the terminal and run `npx react-native run-ios` again. At this point, when tapping on the button, you should see this image:
+
+<figure>
+  <img src="docs/assets/Turbomodules/OldArchAppAlert.png" />
+  <figcaption>The simulator using the new module</figcaption>
+</figure>
+
+Congratulation! The Native Module is now correctly connected to your app.
+If you now relaunch the app, you should see the status getting updated.
+
+:::warning
+This is an example. Don't use this module for your location-based features.
+:::
+
+## Create the Turbomodule
+
+Let's now see how to create a Turbomodule. A Turbomodule is defined by a set of JS spec that needs to be typed. For that reason, we are going to use `Flow` dialect. `TypeScript` support is still in beta but we hope to release it soon.
+
+Once we have the JS spec, we can proceed generating the C++ code that bridges the native code with the JS counterpart.
+
+For the final step, we will implement the missing bits in the native code and then we will connect it to JS.
+
+But first, let's create a testing environment suitable for the new architecture.
+
+### Create the New Architecture Testing Environment
+
+The steps to create a New Architecture testing environment are pretty much the same as [here](#create-a-testing-environment).
+
+There are only two differences. The first one is how we create the app. We will need to run:
+
+```sh
+npx react-native init NewArchitectureApp --version next
+```
+
+The `--version next` bit will let us retrieve a version of React Native that is compatible with the new architecture. We can then proceed with:
+
+```sh
+yarn install
+```
+
+The second difference arrives here, with the pod installation. Instead of running the usual `pod install`, we should run:
+
+```sh
+cd ios && RCT_NEW_ARCH_ENABLED=1 pod install && cd ..
+```
+
+The `RCT_NEW_ARCH_ENABLED` flag will tell to cocoapods to install all the dependencies required by the New Architecture. As final step, let's start Metro in a different terminal:
+
+```sh
+npx react-native start
+```
+
+And let's run the app, to make sure that everything works:
+
+```sh
+npx react-native run-ios
+```
+
+### Adding the JS Spec
+
+Let's add the Turbomodule JS specs. As first thing, we will extract the `AuthorizationStatus` in its own file, to share it between the two modules.
```

Comment:
Couldn't we just let the user create two separate files beforehand?

Reply:
Done! :D